app/src/main/java/org/simple/clinic/summary/PatientSummaryScreen.kt

```diff
@@ -170,6 +170,39 @@ class PatientSummaryScreen(
 
     val screenDestroys: Observable<ScreenDestroyed> = detaches().map { ScreenDestroyed() }
     alertFacilityChangeSheetClosed(screenDestroys)
+    setupChildViewVisibility(screenDestroys)
+  }
+
+  @SuppressLint("CheckResult")
+  private fun setupChildViewVisibility(
+      screenDestroys: Observable<ScreenDestroyed>
+  ) {
+    val modelUpdates: List<Observable<PatientSummaryChildModel>> =
+        listOf(
+            this,
+            drugSummaryView,
+            bloodPressureSummaryView,
+            bloodSugarSummaryView,
+            assignedFacilityView,
+            medicalHistorySummaryView
+        ).map(::createSummaryChildModelStream)
+
+    Observable
+        .combineLatest(modelUpdates) { models -> models.map { it as PatientSummaryChildModel } }
+        .filter { models -> models.all(PatientSummaryChildModel::readyToRender) }
+        .take(1)
+        .takeUntil(screenDestroys)
+        .subscribe { summaryViewsContainer.visibility = VISIBLE }
+  }
+
+  private fun createSummaryChildModelStream(
+      summaryChildView: PatientSummaryChildView
+  ): Observable<PatientSummaryChildModel> {
+    return Observable.create { emitter ->
+      summaryChildView.registerSummaryModelUpdateCallback(emitter::onNext)
```

Comment:
What would be the problem? The view will still be a part of the view hierarchy so it will emit models like any other.

Reply:
Okay, I was just wondering whether it might cause an endless loop. But NVM, forgot delegate automatically starts and stops when attached and detached.